modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/discovery/impl/DiscoveryServiceImpl.java

```diff
@@ -109,4 +100,37 @@ public Response discoverGet() {
         }
         return Response.ok().build();
     }
+
+    /**
+     * Get authentication policy from the tenant configuration, otherwise set default value as Federated.
+     *
+     * @return Authentication policy.
+     * @throws WindowsDeviceEnrolmentException
+     */
+    public String getAuthPolicy() throws WindowsDeviceEnrolmentException {
+        String authPolicy = null;
+        List<ConfigurationEntry> tenantConfigurations;
+        try {
+            if (WindowsAPIUtils.getTenantConfigurationData() != null) {
+                tenantConfigurations = WindowsAPIUtils.getTenantConfigurationData();
+
+                for (ConfigurationEntry configurationEntry : tenantConfigurations) {
+                    if (configurationEntry.getName().equals(PluginConstants.TenantConfigProperties.AUTH_POLICY)) {
+                        authPolicy = configurationEntry.getValue().toString();
+                    } else {
+                        authPolicy = PluginConstants.TenantConfigProperties.DEFAULT_AUTH_POLICY;
+                    }
+                }
+            } else {
+                authPolicy = PluginConstants.TenantConfigProperties.DEFAULT_AUTH_POLICY;
+                String msg = "Tenant configurations are not initialized yet.";
+                log.error(msg);
+            }
+        } catch (DeviceManagementException e) {
+            String msg = "Error occurred in while getting tenant configurations.";
+            log.error(msg);
```

Comment:
(configurationEntry.getName().equals(PluginConstants.TenantConfigProperties.AUTH_POLICY)) 

As per configurationEntry getName will not be null IMO in this scenario. But when checking equals its always better to add the constant first since it will not give any NPEs.